design-proposals/vm-delete-protection.md

```diff
@@ -0,0 +1,147 @@
+# Overview
+
+VirtualMachine Delete Protection prevents the accidental deletion of VMs from a cluster. As long as deletion protection
+is activated for certain VM objects, they cannot be deleted unless the protection is intentionally removed.
+
+## Motivation
+
+KubeVirt clusters are often managed by automations, CLI commands, 3rd party tools or UI actions, for example to deploy,
+reconfigure or delete VMs. These automations may result in accidental deletion of VMs that should have not been deleted.
+Affected VMs could be critical to application/s the cluster is running, or they may contain crucial data. Therefore, the
+VM deletion protection aims to avoid accidental degradation of applications or loss of data.
+
+## Goals
+
+- Establish a delete protection mechanism for VirtualMachine objects.
+
+## Non Goals
+
+- Create a delete protection for VirtualMachineInstance objects.
+- Create a delete protection for virt-launcher Pod objects.
+
+## Definition of Users
+
+Everyone who has the permissions to delete VMs.
+
+## User Stories
+
+As a user who has the permissions to delete VMs, I want to protect my VMs from accidental deletion.
+
+## Repos
+
+kubevirt/kubevirt
+
+# Design
+
+The design is divided into two sections. The first section shows how the VM protection can be enabled by a user, the
+second section depicts possible implementations of the feature in the backend.
+
+### Enable the VM Object Protection
+
+To protect VMs against deletion, they need to be annotated with  `kubevirt.io/vm-delete-protection` set to `true` or
+`True`. If the annotation is present with one of the aforementioned values, requests to delete this VM will be rejected
+by KubeVirt. By default, this annotation is not present on a VM. It is up to the user to enable the deletion protection.
+The annotation looks like this in the VM manifests:
+
+```yaml
+apiVersion: kubevirt.io/v1
+kind: VirtualMachine
+metadata:
+  annotations:
+    kubevirt.io/vm-delete-protection: "true"
+spec:
+[...]
+```
+
+### Backend
+
+#### Option \#1: ValidatingAdmissionPolicy
+
+The VM protection is implemented by adding
+a  [ValidationAdmissionPolicy and ValidationAdmissionPolicyBinding](https://kubernetes.io/docs/reference/access-authn-authz/validating-admission-policy/)
+objects. On every delete action against a VM object, the ValidatingAdmissionPolicy will check for presence of the
+annotation. If it is present and its value equals “true” or “True” the request will be rejected. The
```

Comment:
Would it be useful to explain its behavior in scenarios where it is  misconfigured?

Reply:
It might create a really verbose and noisy outputs IMO. Documentation should cover how to properly configure the feature, and it should make clear that other configurations/values, will allow the deletion.